torch/csrc/autograd/profiler.cpp

```diff
@@ -154,8 +154,37 @@ void enableProfiler(ProfilerConfig config) {
         } else {
           pushRangeImpl(fn.name(), msg, fn.seqNr(), {});
         }
+        fn.setThreadId(thread_id);
+      },
+      [](const RecordFunction& fn) {
+        if (fn.overrideThreadId()) {
+          if (state == ProfilerState::Disabled) {
+            return;
+          } else {
+            std::lock_guard<std::mutex> guard(all_event_lists_mutex);
+            for (const auto& elist : all_event_lists) {
```

Comment:
This might be pretty expensive. Can we change all_event_lists to be a map from thread_id to event_list? That way we can quickly look up the appropriate event_list.

Reply:
I think we can do this, although, my reasoning was that there is only going to be one `event_list` per thread, so `all_event_list` can at most be num_threads, which only happens when all threads use `RecordFunction` in some way. I'm not sure if the number of threads we can have can result in this being a bottleneck.

Reply:
Changed it to use unordered_map